File: sdk/go/unified_intelligence.go
package llmengine

import (
	"bytes"
	"context"
	"encoding/json"
	"net/http"
)

type UnifiedIntelligenceService struct {
	client *Client
}

type UnifiedSignalPayload struct {
	Domain             string                 `json:"domain"`
	EntityReference    string                 `json:"entity_reference"`
	SignalType         string                 `json:"signal_type"`
	Severity           string                 `json:"severity"`
	ConfidenceScore    float64                `json:"confidence_score"`
	RiskScore          float64                `json:"risk_score"`
	EvidenceReferences []string               `json:"evidence_references"`
	Metadata           map[string]interface{} `json:"metadata"`
	IdempotencyKey     string                 `json:"idempotency_key,omitempty"`
}

func (s *UnifiedIntelligenceService) IngestSignal(ctx context.Context, tenantID string, payload UnifiedSignalPayload) (map[string]interface{}, error) {
	buf, err := json.Marshal(payload)
	if err != nil {
		return nil, err
	}
	req, err := http.NewRequestWithContext(ctx, "POST", s.client.baseURL+"/v1/intelligence/signals", bytes.NewReader(buf))
	if err != nil {
		return nil, err
	}
	req.Header.Set("Content-Type", "application/json")
	req.Header.Set("x-tenant-id", tenantID)

	res, err := s.client.hc.Do(req)
	if err != nil {
		return nil, err
	}
	defer res.Body.Close()

	var result map[string]interface{}
	err = json.NewDecoder(res.Body).Decode(&result)
	return result, err
}

func (s *UnifiedIntelligenceService) EvaluateSituations(ctx context.Context, tenantID string) ([]map[string]interface{}, error) {
	req, err := http.NewRequestWithContext(ctx, "POST", s.client.baseURL+"/v1/intelligence/situations/evaluate", nil)
	if err != nil {
		return nil, err
	}
	req.Header.Set("x-tenant-id", tenantID)

	res, err := s.client.hc.Do(req)
	if err != nil {
		return nil, err
	}
	defer res.Body.Close()

	var result []map[string]interface{}
	err = json.NewDecoder(res.Body).Decode(&result)
	return result, err
}

func (s *UnifiedIntelligenceService) EvaluateRisk(ctx context.Context, tenantID string) (map[string]interface{}, error) {
	req, err := http.NewRequestWithContext(ctx, "GET", s.client.baseURL+"/v1/intelligence/risk", nil)
	if err != nil {
		return nil, err
	}
	req.Header.Set("x-tenant-id", tenantID)

	res, err := s.client.hc.Do(req)
	if err != nil {
		return nil, err
	}
	defer res.Body.Close()

	var result map[string]interface{}
	err = json.NewDecoder(res.Body).Decode(&result)
	return result, err
}

File: sdk/typescript/src/unified_intelligence.ts
/**
 * TypeScript SDK client for Phase 5.51 Enterprise AI Unified Intelligence Platform.
 */

export interface UnifiedSignalInput {
  domain: string;
  entity_reference: string;
  signal_type: string;
  severity?: string;
  confidence_score?: number;
  risk_score?: number;
  evidence_references?: string[];
  metadata?: Record<string, any>;
  idempotency_key?: string;
}

export class UnifiedIntelligenceClient {
  private baseUrl: string;
  private apiKey?: string;

  constructor(baseUrl: string = "http://localhost:8000", apiKey?: string) {
    this.baseUrl = baseUrl.replace(/\/$/, "");
    this.apiKey = apiKey;
  }

  private getHeaders(tenantId: string = "default_tenant"): Record<string, string> {
    const headers: Record<string, string> = {
      "Content-Type": "application/json",
      "x-tenant-id": tenantId
    };
    if (this.apiKey) {
      headers["Authorization"] = `Bearer ${this.apiKey}`;
    }
    return headers;
  }

  async ingestSignal(input: UnifiedSignalInput, tenantId: string = "default_tenant"): Promise<any> {
    const res = await fetch(`${this.baseUrl}/v1/intelligence/signals`, {
      method: "POST",
      headers: this.getHeaders(tenantId),
      body: JSON.stringify(input)
    });
    return res.json();
  }

  async evaluateSituations(tenantId: string = "default_tenant"): Promise<any[]> {
    const res = await fetch(`${this.baseUrl}/v1/intelligence/situations/evaluate`, {
      method: "POST",
      headers: this.getHeaders(tenantId)
    });
    return res.json();
  }

  async evaluateRisk(tenantId: string = "default_tenant"): Promise<any> {
    const res = await fetch(`${this.baseUrl}/v1/intelligence/risk`, {
      method: "GET",
      headers: this.getHeaders(tenantId)
    });
    return res.json();
  }

  async evaluateAssurance(tenantId: string = "default_tenant"): Promise<any> {
    const res = await fetch(`${this.baseUrl}/v1/intelligence/assurance`, {
      method: "GET",
      headers: this.getHeaders(tenantId)
    });
    return res.json();
  }

  async evaluateTrust(entityRef: string, tenantId: string = "default_tenant"): Promise<any> {
    const res = await fetch(`${this.baseUrl}/v1/intelligence/trust/${encodeURIComponent(entityRef)}`, {
      method: "GET",
      headers: this.getHeaders(tenantId)
    });
    return res.json();
  }

  async generateRecommendations(situationId: string, tenantId: string = "default_tenant"): Promise<any[]> {
    const res = await fetch(`${this.baseUrl}/v1/intelligence/recommendations/${situationId}`, {
      method: "POST",
      headers: this.getHeaders(tenantId)
    });
    return res.json();
  }

  async getAnalytics(tenantId: string = "default_tenant"): Promise<any> {
    const res = await fetch(`${this.baseUrl}/v1/intelligence/analytics`, {
      method: "GET",
      headers: this.getHeaders(tenantId)
    });
    return res.json();
  }

  async getBilling(tenantId: string = "default_tenant"): Promise<any> {
    const res = await fetch(`${this.baseUrl}/v1/intelligence/billing`, {
      method: "GET",
      headers: this.getHeaders(tenantId)
    });
    return res.json();
  }
}
